app/src/main/kotlin/com/wire/android/ui/home/conversations/details/GroupConversationDetailsViewModel.kt

```diff
@@ -79,51 +81,60 @@ class GroupConversationDetailsViewModel @Inject constructor(
-                                .map { it.conversationDetails }
-                            combine(
-                                conversationDetailsFlow,
-                                isSelfAdminFlow,
-                                getSelfTeam()
-                            ) { conversationDetails, isSelfAnAdmin, selfTeam ->
-                                Triple(
-                                    conversationDetails,
-                                    isSelfAnAdmin,
-                                    selfTeam
-                                )
-                            }.collect { (conversationDetails, isSelfAnAdmin, selfTeam) ->
-                                with(conversationDetails) {
-                                    val isSelfInOwnerTeam =
-                                        selfTeam?.id != null && selfTeam.id == conversationDetails.conversation.teamId?.value
+        viewModelScope.launch(dispatcher.io()) {
+
+            val groupDetailsFlow = observeConversationDetails(conversationId)
+                .filterIsInstance<ObserveConversationDetailsUseCase.Result.Success>()
+                .map { it.conversationDetails }
+                .filterIsInstance<ConversationDetails.Group>()
+                .distinctUntilChanged()
+                .shareIn(this, SharingStarted.WhileSubscribed(), 1)
+
+            val isSelfAdminFlow = observeConversationMembers(conversationId)
+                .map { it.isSelfAnAdmin }
+                .distinctUntilChanged()
 
-                                    if (this is ConversationDetails.Group) {
-                                        updateState(
-                                            groupOptionsState.copy(
-                                                groupName = conversation.name.orEmpty(),
-                                                protocolInfo = conversation.protocol,
-                                                isUpdatingAllowed = isSelfAnAdmin,
-                                                areAccessOptionsAvailable = conversation.isTeamGroup(),
-                                                isGuestAllowed = (conversation.isGuestAllowed() || conversation.isNonTeamMemberAllowed()),
-                                                isServicesAllowed = conversation.isServicesAllowed(),
-                                                isUpdatingGuestAllowed = isSelfAnAdmin && isSelfInOwnerTeam,
-                                                isAbleToRemoveGroup = selfUser.teamId != null
-                                                        && conversation.creatorId.value == selfUser.id.value
-                                            )
-                                        )
-                                    }
-                                }
-                            }
+            launch {
+                groupDetailsFlow
```

Comment:
Why are we launching twice the `groupDetailsFlow` here and on the next coroutine?

Reply:
one of the ideas of that changes is to display data in portions: when some portion of data is ready - display it. 
So the first usage of `groupDetailsFlow` is for displaying data that depends only on groupDetails. The second usage of `groupDetailsFlow` is for displaying data that depends not only on groupDetails but on some other Flows too. 

So if these other Flows "brakes" us (do not emit anything for a log time) then we display at least something that is already fetched.